lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -154,14 +155,39 @@ def import_associations(entities)
               sub_records = Array(entity.send(assoc.name))
               # Set IDS from master to each of the records in `has_one` or `has_many` relation
               sub_records.each { |d| d.send("#{assoc.foreign_key}=", entity.send(assoc.active_record_primary_key)) }
+
+              # Retrieve bulk_import_id data to deleting repeated lines
+              delete_map = mapping_delete_query(entity, assoc)
+
               sub_records
             }.flatten
 
             columns = key_columns(nil, assoc.klass)
             save_records_to_database(assoc.klass, columns, sub_records) if sub_records.any?
+            removing_repeated_associations(delete_map, columns, assoc.klass)
           end
       end
 
+      def mapping_delete_query(entity, assoc)
+        delete_map = []
+        if entity.attributes['bulk_import_id'].present?
+          delete_map.push('bulk_import_id' => entity.attributes['bulk_import_id'],
```

Comment:
maybe i am misreading something - seems to me `delete_map` is always an array with only 1 element? Any particular reason why we want to do it this way?

Reply:
Good catch, I fixed the logic